chatsky/llm/example_selector.py

```diff
@@ -0,0 +1,85 @@
+"""
+Example selection
+-------------------
+This module provides support for example guided generation.
+"""
+
+import asyncio
+from typing import Any, Dict, List
+
+from langchain_core.example_selectors.base import BaseExampleSelector
+from langchain_core.messages import AIMessage, HumanMessage
+from langchain_core.messages.base import BaseMessage
+from pydantic import BaseModel, Field, RootModel
+
+
+class Example(BaseModel):
+    """Class that holds an example"""
+
+    input: str | BaseModel
+    """input may be in form of a string or a custom pydantic model derived from the BaseModel"""
+    output: str | BaseModel
+    """output may be in form of a string or a custom pydantic model derived from the BaseModel"""
+
+
+async def to_langchain_context(
+    example_selector: BaseExampleSelector, input_variables: Dict[str, str]
+) -> List[BaseMessage]:
+    """
+    Function that selects examples and returns them in the format of a list with langchain messages.
+
+    :param example_selector: selector object that implements selection logic.
+    :param input_variables:  this parameter will be passed to example_selector to provide a way to change its behavior in run-time.
+
+    :return: List of Langchain message objects.
+    
+    """
+
+    result = []
+    for example in await example_selector.aselect_examples(input_variables):
+
+        result.append(HumanMessage(content=example["input"]))
+        result.append(AIMessage(content=example["output"]))
+
+    return result
+
+
+class StaticExampleSelector(BaseExampleSelector, RootModel):
+    """Example selector class that selects all examples it holds in root"""
+    
+    root: List[Example]
+    """Examples that StaticExampleSelector holds"""
```

Comment:
You can add cross-references to other objects with 
```
:obj:`root`
```
This would create a link to the `root` attribute.

https://www.sphinx-doc.org/en/master/usage/domains/python.html#python-xref-roles